src/autoschedulers/anderson2021/Makefile

```diff
@@ -0,0 +1,292 @@
+										$(SRC)/LoopNestParser.h \
+										$(SRC)/GPUMemInfo.h \
+										$(SRC)/GPULoopInfo.h \
+										$(SRC)/GPULoopInfo.cpp \
+										$(SRC)/ThreadInfo.h \
+										$(SRC)/Featurization.h \
+										$(SRC)/CostModel.h \
+										$(SRC)/PerfectHashMap.h \
+										$(SRC)/SearchSpace.h \
+										$(SRC)/SearchSpace.cpp \
+										$(SRC)/SearchSpaceOptions.h \
+										$(SRC)/State.h \
+										$(SRC)/State.cpp \
+										$(SRC)/Statistics.h \
+										$(SRC)/Tiling.h \
+										$(SRC)/Tiling.cpp \
+										$(AUTOSCHED_WEIGHT_OBJECTS) \
+										$(AUTOSCHED_COST_MODEL_LIBS) \
+										$(GENERATOR_DEPS) \
+										$(BIN)/auto_schedule_runtime.a
+	@mkdir -p $(@D)
+	$(CXX) -shared $(USE_EXPORT_DYNAMIC) -fPIC -fvisibility=hidden -fvisibility-inlines-hidden $(CXXFLAGS) $(OPTIMIZE) -I $(BIN)/cost_model $(filter-out %.h $(LIBHALIDE_LDFLAGS),$^) -o $@ $(HALIDE_SYSTEM_LIBS)
+
+$(BIN)/retrain_cost_model: $(SRC)/retrain_cost_model.cpp \
+									$(SRC)/ASLog.cpp \
+									$(SRC)/DefaultCostModel.h \
+									$(SRC)/DefaultCostModel.cpp \
+									$(SRC)/Weights.h \
+									$(SRC)/Weights.cpp \
+									$(SRC)/CostModel.h \
+									$(SRC)/NetworkSize.h \
+									$(AUTOSCHED_COST_MODEL_LIBS) \
+									$(AUTOSCHED_WEIGHT_OBJECTS) \
+									$(BIN)/auto_schedule_runtime.a
+	@mkdir -p $(@D)
+	$(CXX) $(CXXFLAGS) -frtti -Wall -I ../support -I $(BIN)/cost_model $(OPTIMIZE) $(filter-out %.h,$^) -o $@ $(LIBHALIDE_LDFLAGS) $(USE_OPEN_MP)
+
+$(BIN)/featurization_to_sample: $(SRC)/featurization_to_sample.cpp
+	@mkdir -p $(@D)
+	$(CXX) $(CXXFLAGS) $< $(OPTIMIZE) -o $@
+
+$(BIN)/get_host_target: $(SRC)/get_host_target.cpp $(LIB_HALIDE) $(HALIDE_DISTRIB_PATH)/include/Halide.h
+	@mkdir -p $(@D)
+	$(CXX) $(CXXFLAGS) $(filter %.cpp,$^) $(LIBHALIDE_LDFLAGS) $(OPTIMIZE) -o $@
+
+$(BIN)/weightsdir_to_weightsfile: $(SRC)/weightsdir_to_weightsfile.cpp $(SRC)/Weights.cpp
+	@mkdir -p $(@D)
+	$(CXX) $(CXXFLAGS) $^ $(OPTIMIZE) -o $@
+
+# This is the value that machine_params defaults to if no custom value is specified;
+# see MachineParams::generic()
+HL_MACHINE_PARAMS ?= 80,25165824,160
+
+# A sample generator to autoschedule. Note that if it statically links
+# to libHalide, then it must be build with $(USE_EXPORT_DYNAMIC), or the
+# autoscheduler can't find the libHalide symbols that it needs.
+$(GENERATOR_BIN)/demo.generator: demo_generator.cpp $(GENERATOR_DEPS)
+	@mkdir -p $(@D)
+	$(CXX) $(CXXFLAGS) $(USE_EXPORT_DYNAMIC) -g $(filter %.cpp,$^) -o $@ $(LIBHALIDE_LDFLAGS)
+
+# To use the autoscheduler, set a few environment variables and use the -p flag to the generator to load the autoscheduler as a plugin
+$(BIN)/%/demo.a: $(GENERATOR_BIN)/demo.generator $(BIN)/libautoschedule_anderson2021.$(SHARED_EXT)
+	@mkdir -p $(@D)
+	HL_WEIGHTS_DIR=$(SRC)/baseline.weights \
+	$(GENERATOR_BIN)/demo.generator -g demo -o $(@D) -f demo target=$* auto_schedule=true -p $(BIN)/libautoschedule_anderson2021.$(SHARED_EXT) -s Anderson2021
```

Comment:
Here and elsewhere: `auto_schedule=true -s Anderson2021` is no longer supported syntax. Instead, specify `autoscheduler=Anderson2021` to replace both of these.